source/slang/slang-parameter-binding.cpp

```diff
@@ -329,11 +329,18 @@ struct UsedRanges
     }
 };
 
+struct ParameterAndLayoutConstants
+{
+    static const uint32_t kInputAttachmentIndexUnused = 0xFFFFFFFF;
+};
+
 struct ParameterBindingInfo
 {
     size_t              space = 0;
     size_t              index = 0;
     LayoutSize          count = 0;
+
+    uint32_t            inputAttachmentIndex = ParameterAndLayoutConstants::kInputAttachmentIndexUnused;
 };
 
 struct ParameterBindingAndKindInfo : ParameterBindingInfo
```

Comment:
I am not convinced that you need to add a new member variable here.
Because when I look at ParameterBindingAndKindInfo, your new member variable will be added for all of the kinds.
Is it what you need?
Shouldn't you just add a new kind to "SlangParameterCategory" and use the existing member variables?

Reply:
We need a new member or some form of extension value to handle subpass's since we have the unique case of 3 parameter binding information's which may conflict.

Subpass's are composed of a `set`+`binding`+`input_attachment_index`. Every single other global parameter type has either: `layout`+`set`, `binding`+`set`, or some variation, allowing `space`+`index` to handle all cases. This is not the case for Subpass.